substrate/frame/nomination-pools/src/lib.rs

```diff
@@ -3263,31 +3349,40 @@ impl<T: Config> Pallet<T> {
 }
 
 impl<T: Config> sp_staking::OnStakingUpdate<T::AccountId, BalanceOf<T>> for Pallet<T> {
+	/// Reduces the balances of the [`SubPools::with_era`], that belong to the pool involved in the
+	/// slash, to the amount that is defined in the `slashed_unlocking` field of
+	/// [`sp_staking::OnStakingUpdate::on_slash`]
+	///
+	/// Emits [`Event::<T>::PoolsSlashed`].
 	fn on_slash(
 		pool_account: &T::AccountId,
 		// Bonded balance is always read directly from staking, therefore we don't need to update
 		// anything here.
 		slashed_bonded: BalanceOf<T>,
 		slashed_unlocking: &BTreeMap<EraIndex, BalanceOf<T>>,
+		total_slashed: BalanceOf<T>,
```

Comment:
The total slashed can be calculated from slashed_bonded and slashed_unlocking right? We could probably get rid of `slashed_bonded ` and emit `total_slashed` in the PoolSlashed event.

Reply:
This would be possible but might also be a little confusing as there are already distinct events for a slash towards a `BondedPool` and an `UnbondingPool`. Having in mind that these events are emitted it would make a lot of sense to either stick to the way it is now or unite both events into one and have just `total_slashed` be included into the single event.